packages/transaction-controller/src/TransactionController.ts

```diff
@@ -285,43 +319,238 @@ const controllerName = 'TransactionController';
 /**
  * The external actions available to the {@link TransactionController}.
  */
-type AllowedActions =
+export type AllowedActions =
   | AddApprovalRequest
   | NetworkControllerFindNetworkClientIdByChainIdAction
   | NetworkControllerGetNetworkClientByIdAction;
 
-type AllowedEvents = NetworkControllerStateChangeEvent;
+/**
+ * The external events available to the {@link TransactionController}.
+ */
+export type AllowedEvents = NetworkControllerStateChangeEvent;
+
+/**
+ * Represents the `TransactionController:stateChange` event.
+ */
+export type TransactionControllerStateChangeEvent = ControllerStateChangeEvent<
+  typeof controllerName,
+  TransactionControllerState
+>;
+
+/**
+ * Represents the `TransactionController:incomingTransactionBlockReceived` event.
+ */
+export type TransactionControllerIncomingTransactionBlockEvent = {
+  type: `${typeof controllerName}:incomingTransactionBlock`;
+  payload: [blockNumber: number];
+};
```

Comment:
```suggestion
/**
 * Represents the `TransactionController:incomingTransactionBlockReceived` event.
 */
export type TransactionControllerIncomingTransactionBlockReceivedEvent = {
  type: `${typeof controllerName}:incomingTransactionBlockReceived`;
  payload: [blockNumber: number];
};
```

Reply:
or maybe just the comment needs to be updated instead

Reply:
Ah, good catch. I think I had renamed this event `incomingTransactionBlockReceived` and decided to revert it. However, now that I think about it, this is a better name. Fixed here: [55dd546](https://github.com/MetaMask/core/pull/3827/commits/55dd546e6d0143fffa04861b107f80d4fe92c723)